src/pymor/models/iosys.py

```diff
@@ -147,6 +165,26 @@ def __init__(self, A, B, C, D=None, E=None, sampling_time=0, presets=None,
         sampling_time = float(sampling_time)
         assert sampling_time >= 0
 
+        assert T is None or T > 0
+
+        if T is not None:
+            if initial_data is None:
+                initial_data = A.source.zeros(1)
+            if isinstance(initial_data, VectorArray):
+                assert initial_data in A.source
+                assert len(initial_data) == 1
+                initial_data = VectorOperator(initial_data, name='initial_data')
+            assert initial_data.source.is_scalar
+            assert initial_data.range == A.source
```

Comment:
I think `initial_data` should also be taken care of when `T` is `None`?

Reply:
The interpretation I used is that, if `T` is `None`, the user doesn't want to do time-stepping, and the initial condition doesn't have to be set up.

Reply:
Yes that's fine. But I still think that we should do some consistency check. Allowing to pass arbitrary stuff for `initial_data` does not feel right to me. I'd be either fine with doing this stuff whenever `initial_data` is not `None` or force `initial_data` to be `None` whenever `T` is `None`.

Reply:
An exception is now raised.